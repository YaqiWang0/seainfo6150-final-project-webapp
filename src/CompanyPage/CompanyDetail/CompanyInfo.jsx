import React, {useState, useEffect} from "react"
import style from "./CompanyInfo.module.css"

const CompanyInfo = (props) => {
    const [subscribe, setSubscribe] = useState(false);
    let sub = ""
    if(props.login.userId !== undefined && props.login.userId !== "") {
        props.firebase.user(props.login.userId).on('value',(snapshot) => {
            sub = snapshot.val();
        })
    }



    const click = () => {
        if(subscribe === false) {
            props.firebase.user(props.login.userId).child(props.info[0].symbol).set("sub")

        } else {
            props.firebase.user(props.login.userId).child(props.info[0].symbol).remove();
        }
        setSubscribe(!subscribe)
    }

    return (
        <div className={style.containerTotal}>
           <div className={style.textInfo}>
               <div className={style.title}>{props.info[0].companyName}</div>
               <div className={style.sub}>
                   <div className={style.stockPriceBlock}>
                       <div className={props.info[0].changes >= 0 ? style.price_red : style.price_green}>{`$${props.info[0].price}`}</div>
                       <div className={props.info[0].changes >= 0 ? style.stock_red : style.stock_green}>{props.info[0].changes >=0 ? `(+ ${props.info[0].changes})` : `( ${props.info[0].changes})`}</div>
                   </div>
                   <div className={props.login.userId === undefined || props.login.userId === "" ? style.hide : style.show}>
                   <div onClick={click} className={sub !== null && sub !== "" && sub[props.info[0].symbol] !== undefined ? style.heartSub : style.heartUnSub}></div>
                       </div>
               </div>
           </div>
            <br/>
            <div className={style.container}>
              <img src={props.info[0].image} alt = {props.info[0].companyName} className={style.pic}/>
                <div className={style.text_block}>
                    <div className={style.quote_left}>"</div>
                    <div className={style.des}>{props.info[0].description}</div>
                    <div className={style.quote_right}>"</div>
                </div>
            </div>
        </div>

    )
}
export default CompanyInfo
