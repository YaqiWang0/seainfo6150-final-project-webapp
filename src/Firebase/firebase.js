import app from 'firebase/app'
import 'firebase/auth'
import 'firebase/database'

const config = {
    apiKey: "AIzaSyCBcPf2qCe4gcUgEeZBpdowfPs_l5HDNsI",
    authDomain: "fall2020-6bd75.firebaseapp.com",
    databaseURL: "https://fall2020-6bd75.firebaseio.com",
    projectId: "fall2020-6bd75",
    storageBucket: "fall2020-6bd75.appspot.com",
    messagingSenderId: "671808993553",
    appId: "1:671808993553:web:87adf307e12583f441f4b0",
    measurementId: "G-79CEYSDSE9"
};

class Firebase {
    constructor() {
        app.initializeApp(config);

        this.auth = app.auth();
        this.db = app.database();
    }


    doCreateUserWithEmailAndPassword = (email, password) =>
        this.auth.createUserWithEmailAndPassword(email, password);

    doSignInWithEmailAndPassword = (email, password) =>
        this.auth.signInWithEmailAndPassword(email, password);

    doSignOut = () => this.auth.signOut();

    user = (uid) => this.db.ref(`users/${uid}`);

    users = () => this.db.ref('users');

}

export default Firebase;
