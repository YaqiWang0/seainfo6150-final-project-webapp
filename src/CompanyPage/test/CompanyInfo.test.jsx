import React from "react";
import data from "./data"
import CompanyInfo from "../CompanyDetail/CompanyInfo";
import Firebase from "../../Firebase/firebase";

describe("CompanyInfo test", () => {
    const login = {
        login: true,
        userId: "NOFjyl43m1Xy3rB4umSRERq8M2z1",
        email: "123456@test.com"
    }

    it('render successfuLly', function () {
        let firebase = new Firebase();
        const {container} = render(
                <CompanyInfo info = {data} firebase = {firebase} login = {login}/>
        );
        expect(container).toMatchSnapshot();
    })
})