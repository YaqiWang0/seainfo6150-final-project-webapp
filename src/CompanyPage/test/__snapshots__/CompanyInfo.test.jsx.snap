// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyInfo test render successfuLly 1`] = `
<div>
  <div
    class="containerTotal"
  >
    <div
      class="textInfo"
    >
      <div
        class="title"
      >
        Apple Inc
      </div>
      <div
        class="sub"
      >
        <div
          class="stockPriceBlock"
        >
          <div
            class="price_green"
          >
            $122.41
          </div>
          <div
            class="stock_green"
          >
            ( -0.83)
          </div>
        </div>
        <div
          class="show"
        >
          <div
            class="heartUnSub"
          />
        </div>
      </div>
    </div>
    <br />
    <div
      class="container"
    >
      <img
        alt="Apple Inc"
        class="pic"
        src="https://financialmodelingprep.com/image-stock/AAPL.png"
      />
      <div
        class="text_block"
      >
        <div
          class="quote_left"
        >
          "
        </div>
        <div
          class="des"
        >
          Apple Inc. designs, manufactures, and markets smartphones, personal computers, tablets, wearables, and accessories worldwide. It also sells various related services. The company offers iPhone, a line of smartphones; Mac, a line of personal computers; iPad, a line of multi-purpose tablets; and wearables, home, and accessories comprising AirPods, Apple TV, Apple Watch, Beats products, HomePod, iPod touch, and other Apple-branded and third-party accessories. It also provides AppleCare support services; cloud services store services; and operates various platforms, including the App Store, that allow customers to discover and download applications and digital content, such as books, music, video, games, and podcasts. In addition, the company offers various services, such as Apple Arcade, a game subscription service; Apple Music, which offers users a curated listening experience with on-demand radio stations; Apple News+, a subscription news and magazine service; Apple TV+, which offers exclusive original content; Apple Card, a co-branded credit card; and Apple Pay, a cashless payment service, as well as licenses its intellectual property. The company serves consumers, and small and mid-sized businesses; and the education, enterprise, and government markets. It sells and delivers third-party applications for its products through the App Store. The company also sells its products through its retail and online stores, and direct sales force; and third-party cellular network carriers, wholesalers, retailers, and resellers. Apple Inc. was founded in 1977 and is headquartered in Cupertino, California.
        </div>
        <div
          class="quote_right"
        >
          "
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CompanyInfo test when the item is not clicked 1`] = `
<div>
  <div
    class="containerTotal"
  >
    <div
      class="textInfo"
    >
      <div
        class="title"
      >
        Apple Inc
      </div>
      <div
        class="sub"
      >
        <div
          class="stockPriceBlock"
        >
          <div
            class="price_green"
          >
            $122.41
          </div>
          <div
            class="stock_green"
          >
            ( -0.83)
          </div>
        </div>
        <div
          class="show"
        >
          <div
            class="heartUnSub"
          />
        </div>
      </div>
    </div>
    <br />
    <div
      class="container"
    >
      <img
        alt="Apple Inc"
        class="pic"
        src="https://financialmodelingprep.com/image-stock/AAPL.png"
      />
      <div
        class="text_block"
      >
        <div
          class="quote_left"
        >
          "
        </div>
        <div
          class="des"
        >
          Apple Inc. designs, manufactures, and markets smartphones, personal computers, tablets, wearables, and accessories worldwide. It also sells various related services. The company offers iPhone, a line of smartphones; Mac, a line of personal computers; iPad, a line of multi-purpose tablets; and wearables, home, and accessories comprising AirPods, Apple TV, Apple Watch, Beats products, HomePod, iPod touch, and other Apple-branded and third-party accessories. It also provides AppleCare support services; cloud services store services; and operates various platforms, including the App Store, that allow customers to discover and download applications and digital content, such as books, music, video, games, and podcasts. In addition, the company offers various services, such as Apple Arcade, a game subscription service; Apple Music, which offers users a curated listening experience with on-demand radio stations; Apple News+, a subscription news and magazine service; Apple TV+, which offers exclusive original content; Apple Card, a co-branded credit card; and Apple Pay, a cashless payment service, as well as licenses its intellectual property. The company serves consumers, and small and mid-sized businesses; and the education, enterprise, and government markets. It sells and delivers third-party applications for its products through the App Store. The company also sells its products through its retail and online stores, and direct sales force; and third-party cellular network carriers, wholesalers, retailers, and resellers. Apple Inc. was founded in 1977 and is headquartered in Cupertino, California.
        </div>
        <div
          class="quote_right"
        >
          "
        </div>
      </div>
    </div>
  </div>
</div>
`;
